File: scripts/create-test-users.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import sys
import io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')
"""
Script to create test users in Supabase for Netra AI
This script uses the Supabase Admin API to create users with proper roles
"""

import os
import sys
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_KEY")

if not SUPABASE_URL or not SUPABASE_SERVICE_KEY:
    print("❌ Error: SUPABASE_URL and SUPABASE_SERVICE_KEY must be set in .env file")
    sys.exit(1)

# Create Supabase admin client
supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)

# Test users to create
TEST_USERS = [
    {
        "email": "sunaysujsy@gmail.com",
        "password": "patient123",
        "role": "patient",
        "full_name": "Sunay Sujsy",
    },
    {
        "email": "rohithpanduru8@gmail.com",
        "password": "doctor123",
        "role": "doctor",
        "full_name": "Rohith Panduru",
    },
    {
        "email": "sunaypotnuru@gmail.com",
        "password": "admin123",
        "role": "admin",
        "full_name": "Sunay Potnuru",
    },
]

def create_user(email: str, password: str, role: str, full_name: str):
    """Create a user in Supabase with the specified role"""
    try:
        print(f"Creating {role} user: {email}...")
        
        # Create user with Supabase Auth
        response = supabase.auth.admin.create_user({
            "email": email,
            "password": password,
            "email_confirm": True,  # Auto-confirm email
            "user_metadata": {
                "role": role,
                "full_name": full_name,
            }
        })
        
        if response.user:
            user_id = response.user.id
            print(f"  [OK] User created with ID: {user_id}")
            
            # Create profile in appropriate table
            if role == "patient":
                profile_data = {
                    "id": user_id,
                    "user_id": user_id,
                    "full_name": full_name,
                    "email": email,
                }
                supabase.table("profiles_patient").insert(profile_data).execute()
                print(f"  [OK] Patient profile created")
                
            elif role == "doctor":
                profile_data = {
                    "id": user_id,
                    "user_id": user_id,
                    "full_name": full_name,
                    "email": email,
                    "specialization": "General Medicine",
                    "verified": True,  # Auto-verify for testing
                }
                supabase.table("profiles_doctor").insert(profile_data).execute()
                print(f"  [OK] Doctor profile created")
                
            elif role == "admin":
                # Admins typically use doctor profile table
                profile_data = {
                    "id": user_id,
                    "user_id": user_id,
                    "full_name": full_name,
                    "email": email,
                    "specialization": "Administration",
                    "verified": True,
                }
                supabase.table("profiles_doctor").insert(profile_data).execute()
                print(f"  [OK] Admin profile created")
            
            return True
        else:
            print(f"  [FAIL] Failed to create user")
            return False
            
    except Exception as e:
        error_msg = str(e)
        if "already registered" in error_msg.lower() or "already exists" in error_msg.lower():
            print(f"  [WARN] User already exists: {email}")
            return True
        else:
            print(f"  [ERROR] Error: {error_msg}")
            return False

def main():
    print("[*] Creating test users for Netra AI...\n")
    print(f"Supabase URL: {SUPABASE_URL}\n")
    
    success_count = 0
    for user in TEST_USERS:
        if create_user(
            email=user["email"],
            password=user["password"],
            role=user["role"],
            full_name=user["full_name"]
        ):
            success_count += 1
        print()
    
    print("=" * 60)
    print(f"[OK] Successfully created/verified {success_count}/{len(TEST_USERS)} users\n")
    print("[*] Test User Credentials:")
    print("-" * 60)
    for user in TEST_USERS:
        print(f"{user['role'].upper():8} | Email: {user['email']:30} | Password: {user['password']}")
    print("-" * 60)
    print("\n[*] Login URLs:")
    print(f"  Patient: http://localhost:5173/login/patient")
    print(f"  Doctor:  http://localhost:5173/login/doctor")
    print(f"  Admin:   http://localhost:5173/login/admin")
    print("\n[OK] You can now login with these credentials!")

if __name__ == "__main__":
    main()

File: scripts/create-test-users.js
#!/usr/bin/env node
/**
 * Script to create test users in Supabase for Netra AI
 * This script uses the Supabase Admin API to create users with proper roles
 */

const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const SUPABASE_URL = process.env.SUPABASE_URL;
const SUPABASE_SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY;

if (!SUPABASE_URL || !SUPABASE_SERVICE_KEY) {
  console.error('❌ Error: SUPABASE_URL and SUPABASE_SERVICE_KEY must be set in .env file');
  process.exit(1);
}

// Create Supabase admin client
const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_KEY, {
  auth: {
    autoRefreshToken: false,
    persistSession: false
  }
});

// Test users to create
const TEST_USERS = [
  {
    email: 'patient@test.com',
    password: 'patient123',
    role: 'patient',
    full_name: 'Test Patient',
  },
  {
    email: 'doctor@test.com',
    password: 'doctor123',
    role: 'doctor',
    full_name: 'Dr. Test Doctor',
  },
  {
    email: 'admin@test.com',
    password: 'admin123',
    role: 'admin',
    full_name: 'Admin User',
  },
];

async function createUser(email, password, role, full_name) {
  try {
    console.log(`Creating ${role} user: ${email}...`);
    
    // Create user with Supabase Auth
    const { data, error } = await supabase.auth.admin.createUser({
      email: email,
      password: password,
      email_confirm: true, // Auto-confirm email
      user_metadata: {
        role: role,
        full_name: full_name,
      }
    });
    
    if (error) {
      if (error.message.includes('already registered') || error.message.includes('already exists')) {
        console.log(`  ⚠️  User already exists: ${email}`);
        return true;
      }
      throw error;
    }
    
    if (data.user) {
      const user_id = data.user.id;
      console.log(`  ✅ User created with ID: ${user_id}`);
      
      // Create profile in appropriate table
      if (role === 'patient') {
        const profile_data = {
          id: user_id,
          user_id: user_id,
          full_name: full_name,
          email: email,
        };
        await supabase.from('profiles_patient').insert(profile_data);
        console.log(`  ✅ Patient profile created`);
        
      } else if (role === 'doctor') {
        const profile_data = {
          id: user_id,
          user_id: user_id,
          full_name: full_name,
          email: email,
          specialization: 'General Medicine',
          verified: true, // Auto-verify for testing
        };
        await supabase.from('profiles_doctor').insert(profile_data);
        console.log(`  ✅ Doctor profile created`);
        
      } else if (role === 'admin') {
        // Admins typically use doctor profile table
        const profile_data = {
          id: user_id,
          user_id: user_id,
          full_name: full_name,
          email: email,
          specialization: 'Administration',
          verified: true,
        };
        await supabase.from('profiles_doctor').insert(profile_data);
        console.log(`  ✅ Admin profile created`);
      }
      
      return true;
    } else {
      console.log(`  ❌ Failed to create user`);
      return false;
    }
    
  } catch (error) {
    console.log(`  ❌ Error: ${error.message}`);
    return false;
  }
}

async function main() {
  console.log('🚀 Creating test users for Netra AI...\n');
  console.log(`Supabase URL: ${SUPABASE_URL}\n`);
  
  let success_count = 0;
  for (const user of TEST_USERS) {
    if (await createUser(user.email, user.password, user.role, user.full_name)) {
      success_count++;
    }
    console.log();
  }
  
  console.log('='.repeat(60));
  console.log(`✅ Successfully created/verified ${success_count}/${TEST_USERS.length} users\n`);
  console.log('📝 Test User Credentials:');
  console.log('-'.repeat(60));
  for (const user of TEST_USERS) {
    console.log(`${user.role.toUpperCase().padEnd(8)} | Email: ${user.email.padEnd(25)} | Password: ${user.password}`);
  }
  console.log('-'.repeat(60));
  console.log('\n🌐 Login URLs:');
  console.log('  Patient: http://localhost:3000/login/patient');
  console.log('  Doctor:  http://localhost:3000/login/doctor');
  console.log('  Admin:   http://localhost:3000/login/admin');
  console.log('\n✅ You can now login with these credentials!');
}

main().catch(console.error);
